draft-irtf-cfrg-hpke.md

```diff
@@ -300,23 +304,33 @@ operations, roles, and behaviors of HPKE:
 HPKE variants rely on the following primitives:
 
 * A Key Encapsulation Mechanism (KEM):
-  - GenerateKeyPair(): Generate a key pair (skX, pkX)
-  - DeriveKeyPair(ikm): Derive a key pair `(skX, pkX)` from the byte string `ikm`,
-    where `ikm` SHOULD have at least `Nsk` bytes of entropy (see
-    {{derive-key-pair}} for discussion)
-  - Serialize(pk): Produce a byte string of length `Npk` encoding the
-    public key `pk`
-  - Deserialize(enc): Parse the byte string `enc` of length `Npk` to recover a
-    public key (note: this function can raise an error upon `enc` deserialization failure)
-  - Encap(pk): Generate an ephemeral, fixed-length symmetric key (the KEM shared secret) and
-    a fixed-length encapsulation of that key that can be decapsulated
-    by the holder of the private key corresponding to pk
+  - GenerateKeyPair() (OPTIONAL): Generate a key pair `(skX, pkX)`
+  - DeriveKeyPair(ikm) (OPTIONAL): Derive a key pair `(skX, pkX)` from
+    the byte string `ikm`, where `ikm` SHOULD have at least `Nsk` bytes
+    of entropy (see {{derive-key-pair}} for discussion)
+  - SerializePublicKey(pk): Produce a byte string of length `Npk`
+    encoding the KEM public key `pk`. A KEM MAY provide multiple
+    different equivalent encoding algorithms for public keys, with
+    different output lengths `Npk`.
```

Comment:
Using a single constant to refer to multiple values seems error prone. Can we avoid this?